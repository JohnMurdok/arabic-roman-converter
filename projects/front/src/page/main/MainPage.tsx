import { Container } from '@mui/material';
import CardComponent from '../../component/card/CardComponent';
import ArabicInputComponent from '../../component/input/ArabicInputComponent';
import RomanInputComponent from '../../component/input/RomanInputComponent';

const CARD_TITLE = 'Arabic number to roman number converter';

const MainPage = () => (
    <Container data-testid={'main-page'} maxWidth={'sm'} >
        <CardComponent data-testid={'main-page-card'} title={CARD_TITLE}>
            <ArabicInputComponent data-testid={'main-page-arabic-input'} />
            <RomanInputComponent data-testid={'main-page-roman-input'} />
        </CardComponent>
    </Container>
);

export default MainPage;
