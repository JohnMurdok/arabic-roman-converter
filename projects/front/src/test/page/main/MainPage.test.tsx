import { render, screen } from '@testing-library/react';
import MainPage from '../../../page/main/MainPage';
jest.mock('../../../component/card/CardComponent', () => 'card-component');
jest.mock('../../../component/input/ArabicInputComponent', () => 'arabic-input-component');
jest.mock('../../../component/input/RomanInputComponent', () => 'roman-input-component');

describe('[PAGE] MainPage', () => {
    test('matches snapshot', async () => {
        render(<MainPage />);
        const mainPageComponent = await screen.findByTestId('main-page');
        expect(mainPageComponent).toMatchSnapshot()
    });

    test('CardComponent should have a title property', async () => {
        render(<MainPage />);
        const cardComponent = await screen.findByTestId('main-page-card');
        expect(cardComponent).toHaveAttribute('title', 'Arabic number to roman number converter');
    });
});
