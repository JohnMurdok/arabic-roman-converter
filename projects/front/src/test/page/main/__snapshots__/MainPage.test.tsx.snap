// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PAGE] MainPage matches snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  data-testid="main-page"
>
  <card-component
    data-testid="main-page-card"
    title="Arabic number to roman number converter"
  >
    <arabic-input-component
      data-testid="main-page-arabic-input"
    />
    <roman-input-component
      data-testid="main-page-roman-input"
    />
  </card-component>
</div>
`;
